drivers/CMakeLists.txt

```diff
@@ -12,7 +12,7 @@ add_subdirectory(usb_c)
 add_subdirectory_ifdef(CONFIG_ADC adc)
 add_subdirectory_ifdef(CONFIG_AUDIO audio)
 add_subdirectory_ifdef(CONFIG_BBRAM bbram)
-add_subdirectory_ifdef(CONFIG_BOARD_XENVM xen)
+add_subdirectory_ifdef(CONFIG_DT_HAS_XEN_XEN_ENABLED xen)
```

Comment:
uhm, for consistency this should be `add_subdirectory_ifdef(CONFIG_XEN)` (or similar) and then use `CONFIG_DT_HAS_XEN_XEN_ENABLED` in the actual driver (see how this is done for the other drivers).